darkside-tests/src/utils.rs

```diff
@@ -384,3 +388,151 @@ impl TreeState {
         })
     }
 }
+
+/// Basic initialisation of darksidewalletd.
+/// Returns a darkside handler and darkside connector for chain builds.
+/// Generates a genesis block and adds initial treestate.
+pub async fn init_darksidewalletd(
+    set_port: Option<portpicker::Port>,
+) -> Result<(DarksideHandler, DarksideConnector), String> {
+    let handler = DarksideHandler::new(set_port);
+    let server_id = zingoconfig::construct_lightwalletd_uri(Some(format!(
+        "http://127.0.0.1:{}",
+        handler.grpc_port
+    )));
+    let connector = DarksideConnector(server_id);
+
+    // Setup prodedures.  Up to this point there's no communication between the client and the dswd
+    let mut client = connector.get_client().await.unwrap();
+    client.clear_address_utxo(Empty {}).await.unwrap();
+
+    // reset with parameters
+    connector
+        .reset(1, String::from(BRANCH_ID), String::from("regtest"))
+        .await
+        .unwrap();
+
+    // stage genesis block
+    connector
+        .stage_blocks_stream(vec![String::from(crate::constants::GENESIS_BLOCK)])
+        .await?;
+    connector
+        .add_tree_state(constants::first_tree_state())
+        .await
+        .unwrap();
+
+    Ok((handler, connector))
+}
+
+/// Stage and apply a range of blocks and update tree state.
+pub async fn generate_blocks(
+    connector: &DarksideConnector,
+    tree_state: TreeState,
+    current_height: i32,
+    target_height: i32,
+    nonce: i32,
+) -> i32 {
+    let count = target_height - current_height;
+    connector
+        .stage_blocks_create(current_height + 1, count, nonce)
+        .await
+        .unwrap();
+    connector
+        .add_tree_state(TreeState {
+            height: target_height as u64,
+            ..tree_state
+        })
+        .await
+        .unwrap();
+    connector.apply_staged(target_height).await.unwrap();
+    target_height
+}
+
+/// Stage a block and transaction, then update tree state.
```

Comment:
is there a scenario in which updating the tree state is not desirable? If so this could be parameterized

Reply:
Good non-blocking idea.